### DISCUSSION TITLE
Asynchronous scatter

### DISCUSSION THREAD

**Discussion by YarShev** [dask/dask#7154] [Q&A] [answered]
Hi guys,

Is it possible to get `Future` from asynchronous `scatter` but not coroutine? This is needed to continue working with `Future`.

```python
from distributed.client import Client
c = Client()
r = c.scatter(1, asynchronous=True)
r
<coroutine object Client._scatter at 0x000001C38A1DF4C0>
```

Thanks in advance!

**Comment by jsignell** [comment, marked as answer]
You can await it just like a regular coroutine:

```python
future = await c.scatter(1, asynchronous=True)
```

You might be interested in reading more about async: https://docs.python.org/3/library/asyncio-task.html

**Reply by YarShev** [reply]
Yes, I know that I can await it. I mean `scatter` itself is synchronous operation that returns a future referencing to data. But can I get a future referencing to data immediately while data is being saved in a worker?

**Reply by jsignell** [reply]
Ah I see. I don't think you can.

**Reply by YarShev** [reply]
Okay, thanks!
